src/Utilities/StdHelpers.hpp

```diff
@@ -393,6 +393,22 @@ inline std::array<T, Dim> operator-(const std::array<T, Dim>& rhs) noexcept(
   return result;
 }
 
+/// \ingroup Utilities
+/// \brief Construct an array from an existing array omitting one element
+template <typename T, size_t Dim, size_t ElementToRemove = 0>
```

Comment:
Put the template arguments that can be inferred last.  Remove the default for ElementToRemove because `all_but_specified_element_of(x)` doesn't specify an element.